lib/promoted/ruby/client/validator.rb

```diff
@@ -0,0 +1,31 @@
+module Promoted
+    module Ruby
+      module Client
+        class Validator
+            def self.validate_fields!(obj, obj_name, fields)
+                fields.each {|field|
+                    msg = field[:name] + " is required on " + obj_name
```

Comment:
Perf question - will this create a new string even if there is not an error?

Reply:
I moved it in.